### PR TITLE
[Fix-8367][Master] execute COMPLEMENT_DATA command always missing some ack event

### LINKED ISSUE #8367: [Bug] [server] execute COMPLEMENT_DATA command always missing some ack event
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

when I execute COMPLEMENT_DATA task.
For example, 1 to 20. And then some task instances always missing field value(start_time/execute_path/log_path)
I find this is due to missing ack event, but I don't kown why missing.
image

### What you expected to happen

all task instance should to be create success.

### How to reproduce

Execute task with COMPLEMENT_DATA and date range is longer and bester.

### Anything else

_No response_

### Version

2.0.3

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

This pr will close #8367

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java [hunk 1]
```diff
@@ -480,28 +481,45 @@ private boolean processComplementData() throws Exception {
                     processInstance.getScheduleTime(),
                     complementListDate.toString());
             scheduleDate = complementListDate.get(index + 1);
-            //the next process complement
-            processInstance.setId(0);
+
+        }
+
+        this.taskInstanceHashMap.clear();
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: should remove this code?

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java [hunk 2]
```diff
@@ -480,28 +481,45 @@ private boolean processComplementData() throws Exception {
...
+
+        //the next process complement
+        int create = this.createComplementDataCommand(scheduleDate);
+        if (create > 0) {
+            logger.info("create complement data command successfully.");
```
Comment (caishunfeng) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: It's better to add some identify flag in log, such as id or code.

**Comment by caishunfeng** [comment]
relative pr in dev: #7976

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java (hunk 1 above)
Comment (SbloodyS) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: I'll remove it.

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java (hunk 2 above)
Comment (SbloodyS) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThread.java]: I'll add it.

**Comment by caishunfeng** [comment]
> SonarCloud Quality Gate failed.    Quality Gate failed
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=BUG) [E](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=BUG) [24 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=SECURITY_HOTSPOT) [3 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=CODE_SMELL) [359 Code Smells](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=8407&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=8407&metric=new_coverage&view=list) [15.7% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=8407&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=8407&metric=new_duplicated_lines_density&view=list) [1.1% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=8407&metric=new_duplicated_lines_density&view=list)

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**